packages/expo-updates/ios/EXUpdates/AppLoader/FileDownloader.swift

```diff
@@ -239,7 +239,10 @@ public final class FileDownloader {
 
     do {
       if let extraClientParams = try database.extraParams(withScopeKey: scopeKey) {
-        extraHeaders["Expo-Extra-Params"] = try StringStringDictionarySerializer.serialize(dictionary: extraClientParams)
+        let structuredHeaderDictionary = try StringDictionary(value: extraClientParams.mapValues({ value in
```

Comment:
Any reason we are using our own class to serialize, instead of just leveraging built in JSON serialization code?

Reply:
I think there are also good existing packages for doing HTML escaping/unescaping

Reply:
This is Structured Field Values (`expo-structured-headers`) instead of JSON:
https://github.com/expo/expo/tree/main/packages/expo-structured-headers
https://docs.expo.dev/technical-specs/expo-sfv-0/

Currently, only the Android implementation is complete (parser and serializer). On iOS, only the parser is implemented fully. We've been supplementing it with piecemeal serialization support but only for the types we need (string dictionary before, and string list in this PR). Hesitant to put it in the package until we have parity in support between the platforms.

We use structured field values in most of our headers.

Reply:
As for why not JSON, I'll see if I can dig up that discussion from expo-updates inception.

Reply:
Ok approved! (Please fix SwiftLint...)